src/fitnesse/reporting/history/ExecutionReport.java

```diff
@@ -66,24 +69,35 @@ public int hashCode() {
     return new HashCodeBuilder().append(rootPath).append(version).append(date).hashCode();
   }
 
-  public static ExecutionReport makeReport(String xmlString) throws Exception {
-    Document xmlDocument = XmlUtil.newDocument(xmlString);
+  public static ExecutionReport makeReport(String xmlString) throws InvalidReportException {
+    Document xmlDocument = null;
+    try {
+      xmlDocument = XmlUtil.newDocument(xmlString);
+    } catch (IOException | SAXException e) {
+      throw new InvalidReportException("%s is not a valid execution report", e);
+    }
     Element documentElement = xmlDocument.getDocumentElement();
     String documentNodeName = documentElement.getNodeName();
-    if (documentNodeName.equals("testResults"))
-      return new TestExecutionReport(xmlDocument);
-    else if (documentNodeName.equals("suiteResults"))
-      return new SuiteExecutionReport(xmlDocument);
-    else
-      throw new RuntimeException(String.format("%s is not a valid document element tag for an Execution Report.", documentNodeName));
+    switch (documentNodeName) {
+      case "testResults":
+        return new TestExecutionReport(xmlDocument);
+      case "suiteResults":
+        return new SuiteExecutionReport(xmlDocument);
+      default:
+        throw new InvalidReportException(String.format("%s is not a valid document element tag for an Execution Report.", documentNodeName));
+    }
   }
 
-  protected void unpackCommonFields(Element documentElement) {
+  protected void unpackCommonFields(Element documentElement) throws InvalidReportException {
     version = XmlUtil.getTextValue(documentElement, "FitNesseVersion");
     rootPath = XmlUtil.getTextValue(documentElement, "rootPath");
     String dateString = XmlUtil.getTextValue(documentElement, "date");
     if (dateString != null)
-      date = DateTimeUtil.getDateFromString(dateString);
+      try {
+        date = DateTimeUtil.getDateFromString(dateString);
+      } catch (ParseException e) {
+        throw new InvalidReportException(String.format("'%s' is not a valid date.", dateString), e);
```

Comment:
If you will be replacing %s more often in the message of exceptions of this type, maybe the String.format() can be part of (an extra) constructor for the exception?